### PR TITLE
Register Node Id and Volume Information in ETCD

### PR DESCRIPTION
**Pull request by rogercloud**
1. Refactor etcd client wrapper and error handling to `common` module.
2. Register Node Id and Volume Information in ETCD.

### CODE REVIEW

File: async_fuse/src/main.rs [hunk 1]
```diff
@@ -44,10 +44,146 @@ use log::debug;
...
+/// ETCD ndoe id info
+const ETCD_NODE_ID_INFO_PREFIX: &str = "datenlord_etcd_node_id_info_";
+/// ETCD volume information lock
+const ETCD_VOLUME_INFO_LOCK: &str = "datenlord_etcd_volume_info_lock";
+/// ETCD volume information prefix
+const ETCD_VOLUME_INFO_PREFIX: &str = "datenlord_etcd_volume_info_";
+
+/// Register current node to etcd and get a dedicated node id.
+/// The registered information contains IP.
+async fn register_node_id(etcd_client: &EtcdDelegate) -> anyhow::Result<u64> {
```
Comment (pzheng1025) [inline async_fuse/src/main.rs:73]: Is this overlapped with https://github.com/datenlord/datenlord/blob/master/csi/src/meta_data.rs#L139?

**Comment by pzheng1025** [comment]
Later csi and fuse will merge together. We'd better to have a signal place to register information to etcd.

File: async_fuse/src/main.rs (hunk 1 above)
Comment (rogercloud) [reply async_fuse/src/main.rs:75]: When will that function be called? And this function also register an IP.

Comment (pzheng1025) [reply async_fuse/src/main.rs:75]: That is called when the pod is created.

**Review by pzheng1025 — APPROVED** [review]

**Review by pwang7 — APPROVED** [review]

**Pull request merged**